src/OrbitGl/TimeGraph.cpp

```diff
@@ -89,18 +89,30 @@ float TimeGraph::GetHeight() const {
 void TimeGraph::UpdateCaptureMinMaxTimestamps() {
   auto [tracks_min_time, tracks_max_time] = GetTrackManager()->GetTracksMinMaxTimestamps();
 
-  capture_min_timestamp_ = std::min(capture_min_timestamp_, tracks_min_time);
-  capture_max_timestamp_ = std::max(capture_max_timestamp_, tracks_max_time);
+  uint64_t capture_min_timestamp = std::min(capture_min_timestamp_, tracks_min_time);
+  uint64_t capture_max_timestamp = std::max(capture_max_timestamp_, tracks_max_time);
+
+  capture_min_timestamp =
+      std::min(capture_min_timestamp, capture_data_->GetCallstackData().min_time());
+  capture_max_timestamp =
+      std::max(capture_max_timestamp, capture_data_->GetCallstackData().max_time());
+
+  if (capture_min_timestamp == capture_min_timestamp_ &&
+      capture_max_timestamp == capture_max_timestamp_)
+    return;
```

Comment:
Why this early return?

Isn't easier to understand:
1- get TrackMinMaxTimestamps
2- make min/max with the member variable
3- make another min with CallstackData and also a max.